### PR TITLE
  Bug 1221427 - ecape key support in smartlist

### PR DESCRIPTION
**Pull request by danhuang1202**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=735b824abff7bf3adf582f3b8cd5613b95154dbc

File: tv_apps/browser/js/smartList.js [hunk 1]
```diff
@@ -670,6 +663,35 @@
...
+        (listData) => {
+          // if user is browser list data in a folder,
+          // add back button data at the first place of listData
+          if(this.navState) {
+            listData.unshift(
+              this.generateBackButtonData(this.navState.folderTitle)
+            );
+          }
+
+          if(listData.length === 0) {
```
Comment (weilonge) [inline tv_apps/browser/js/smartList.js]: nit: add space between 'if ('

File: tv_apps/browser/js/smartList.js [hunk 2]
```diff
@@ -670,6 +663,35 @@
...
+            return;
+          }
+
+          this.render(listData);
+
+          // focus the first item in list. if the first item is button and list
+          // item length is bigger than 1, focus the second item
+          this.focusIndex = 0;
+          if(this.navState && this.listIndexEndAt > 0) {
```
Comment (weilonge) [inline tv_apps/browser/js/smartList.js]: nit: add space between 'if ('

File: tv_apps/browser/js/smartList.js [hunk 3]
```diff
@@ -871,11 +857,34 @@
       }
     },
 
+    handleKeyEscape: function() {
+      if(!this.navState) {
```
Comment (weilonge) [inline tv_apps/browser/js/smartList.js]: nit: add space between 'if (

File: tv_apps/browser/js/smartList.js [hunk 4]
```diff
@@ -670,6 +663,35 @@
       };
     },
 
+    switchListView: function(folderId) {
+      this.dispatchLoadDataByRange(0, MAX_VISIBLE_ITEM*2, folderId,
+        (listData) => {
+          // if user is browser list data in a folder,
+          // add back button data at the first place of listData
+          if(this.navState) {
```
Comment (weilonge) [inline tv_apps/browser/js/smartList.js]: nit: add space between 'if ('

File: tv_apps/browser/js/smartList.js [hunk 5]
```diff
@@ -135,27 +136,19 @@
...
-        number: MAX_VISIBLE_ITEM*2,
-        folderId: null,
-        callback: (function(listData) {
-          var event = new Event('open'),
-              focusEl = null;
+      this.reset();
+      this.dispatchLoadDataByRange(0, MAX_VISIBLE_ITEM*2, null, (listData) => {
+        this.el.classList.add('show');
 
+        if(listData.length > 0) {
```
Comment (weilonge) [inline tv_apps/browser/js/smartList.js]: nit: add space between 'if ('

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bd5f25a18defff76f036890d1a3835bb861913e8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b8440df2fd3e693ea78834a45e1112359303af78

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b6aac7aa647d6f65c7634e3946b77e600613f5a3

**Pull request merged**